mlos_bench/mlos_bench/optimizers/convert_configspace.py

```diff
@@ -8,24 +8,38 @@
 
 import logging
 
-from typing import Optional
-
-from ConfigSpace.hyperparameters import Hyperparameter
-from ConfigSpace import UniformIntegerHyperparameter
-from ConfigSpace import UniformFloatHyperparameter
-from ConfigSpace import CategoricalHyperparameter
-from ConfigSpace import ConfigurationSpace, Configuration
-
-from mlos_bench.tunables.tunable import Tunable
+from typing import Dict, Optional, Tuple
+
+from ConfigSpace import (
+    CategoricalHyperparameter,
+    Configuration,
+    ConfigurationSpace,
+    EqualsCondition,
+    UniformFloatHyperparameter,
+    UniformIntegerHyperparameter,
+)
+from mlos_bench.tunables.tunable import Tunable, TunableValue
 from mlos_bench.tunables.tunable_groups import TunableGroups
 
 _LOG = logging.getLogger(__name__)
 
 
-def _tunable_to_hyperparameter(
-        tunable: Tunable, group_name: Optional[str] = None, cost: int = 0) -> Hyperparameter:
+class TunableValueKind:
+    """
+    Enum for the kind of the tunable value (special or not).
+    It is not a true enum because ConfigSpace wants string values.
     """
-    Convert a single Tunable to an equivalent ConfigSpace Hyperparameter object.
+
+    SPECIAL = "special"
+    RANGE = "range"
+
+
+def _tunable_to_configspace(
+        tunable: Tunable, group_name: Optional[str] = None, cost: int = 0) -> ConfigurationSpace:
+    """
+    Convert a single Tunable to an equivalent set of ConfigSpace Hyperparameter objects,
+    wrapped in a ConfigurationSpace for composability.
+    Note: this may more than one Hyperparameter in the case of special value handling.
```

Comment:
```suggestion
    Note: this may be more than one Hyperparameter in the case of special value handling.
```
correcting my own correction 🤦